src/components/GlobalPreferences/Notifications/NotificationsVerify.js

```diff
@@ -0,0 +1,120 @@
+import React, { useEffect, useState } from 'react'
+import PropTypes from 'prop-types'
+import { ButtonBase, Info, LoadingRing, useTheme } from '@aragon/ui'
+import { verifyEmailToken } from './notification-service-api'
+import {
+  VERIFY_SUBSECTION,
+  UnauthorizedError,
+  ExpiredTokenError,
+} from './constants'
+import NotificationsVerifyBox from './NotificationsVerifyBox'
+
+export function NotificationsVerify({
+  subsection,
+  onTokenChange,
+  onEmailChange,
+  navigateToNotifications,
+}) {
+  const [isFetching, setIsFetching] = useState(true)
+  const [verified, setVerified] = useState(false)
+  const [error, setError] = useState(null)
+  const theme = useTheme()
+
+  useEffect(() => {
+    // Parse token from subsection /verify/[TOKEN] -> [TOKEN]
+    const token = subsection.substring(VERIFY_SUBSECTION.length)
+    verifyEmailToken(token)
+      .then(longLivedToken => {
+        setVerified(true)
+        setIsFetching(false)
+        onTokenChange(longLivedToken)
+        return longLivedToken
+      })
+      .catch(e => {
+        // if an invalid token is passed. we can clear email and token to reset the stae
+        // or present the user with the error and give some options
+        // onEmailChange(null)
+        // onTokenChange(null)
+
+        if (e instanceof ExpiredTokenError) {
+          setError('Your email link has expired.')
+        } else if (e instanceof UnauthorizedError) {
+          setError(
+            "Oops, it looks something went wrong and you weren't authorized. Please try again."
+          )
+        }
+        setIsFetching(false)
+        setVerified(false)
+      })
+  }, [subsection, onTokenChange, onEmailChange])
+
+  if (isFetching) {
+    return (
+      <NotificationsVerifyBox header="Verifying">
```

Comment:
Same comment about alignment. I left a suggestion on how to fix this on `NotificationsVerifyBox`

![image](https://user-images.githubusercontent.com/3072458/63692768-b4cfe200-c812-11e9-826b-10361ee73057.png)